features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Applications/Details/SubscriptionTableData.jsx

```diff
@@ -158,9 +238,103 @@ class SubscriptionTableData extends React.Component {
+                                                />
+                                            )
+                                             : (status === 'REJECTED')
+                                            ? (
+                                                <FormattedMessage
+                                                    id='Applications.Details.SubscriptionTableData.update.throttling.policy.blocked'
+                                                    defaultMessage='Subscription is currently REJECTED state. You need to get approval to the subscription before editing the tier'
+                                                />
+                                            )
+                                             :(   
+                                                <div>
+                                                    <TextField
+                                                        required
+                                                        fullWidth
+                                                        id='outlined-select-currency'
+                                                        select
+                                                        label={(
+                                                            <FormattedMessage
+                                                                defaultMessage='Throttling Tier'
+                                                                id='Applications.Details.SubscriptionTableData.update.throttling.policy.name'
+                                                            />
+                                                        )}
+                                                        value={selectedTier}
+                                                        name='throttlingPolicy'
+                                                        onChange={e => this.setSelectedTier(e.target.value)}
+                                                        helperText={(
+                                                            <FormattedMessage
+                                                                defaultMessage={`Assign a new Throttling policy tier to the existing subscription`}
+                                                                id='Applications.Details.SubscriptionTableData.update.throttling.policy.helper'
+                                                            />
+                                                        )}
+                                                        margin='normal'
+                                                        variant='outlined'
+                                                    >
+                                                        {this.state.tiers.map((tier) => (
+                                                            <MenuItem key={tier.value} value={tier.value}>
```

Comment:
Make sure this key value is a unique value for each. If the key is duplicated, there can be unpredictable issues.

Reply:
Yes, in here, the tier key value pair is unique.